### PR TITLE
Testing in PHP 8.1 + Add test invoke object with magic method `__callStatic()`

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #57

### CODE REVIEW

**Review by roxblnfk — APPROVED** [review]
Can you add test with `static::` call like it does in issue #57 ?

**Comment by vjik** [comment]
> Can you add test with `static::` call like it does in issue #57 ?

This test is fail as described in #57

**Comment by roxblnfk** [comment]
> This test is fail as described in #57

Yes. And i think this case also should be covered (with expected exception)

**Comment by vjik** [comment]
Skip one test because bug in PHP https://bugs.php.net/bug.php?id=81626

**Review by roxblnfk — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

File: tests/Common/InjectorTest.php [hunk 1]
```diff
@@ -67,6 +72,44 @@ public function testInvokeStatic(): void
...
+     */
+    public function testInvokeStaticWithStaticCalls(string $className): void
+    {
+        if (
+            $className === CallStaticWithStaticObject::class
+            && version_compare(PHP_VERSION, '8.1-dev', '<')
+        ) {
+            /** @link https://bugs.php.net/bug.php?id=81626 */
+            $this->markTestSkipped('Bug in PHP version below 8.1. See https://bugs.php.net/bug.php?id=81626');
+        }
```
Comment (roxblnfk) [inline tests/Common/InjectorTest.php:105]: Todo: remove this

Comment (samdark) [reply tests/Common/InjectorTest.php:105]: @roxblnfk it will be lost here. Please create an issue.

**Pull request merged**